roadmap/implementers-guide/src/node/collators/collation-generation.md

```diff
@@ -1,9 +1,36 @@
 # Collation Generation
 
-> TODO
+The collation generation subsystem is executed on collator nodes and produces candidates to be distributed to validators. If configured to produce collations for a para, it produces collations and then feeds them to the [Collator Protocol][CP] subsystem, which handles the networking.
 
 ## Protocol
 
+Input: None
+
+Output: CollationDistributionMessage
+
 ## Functionality
 
-## Jobs, if any
+The process of generating a collation for a parachain is very parachain-specific. As such, the details of how to do so are left beyond the scope of this description. The subsystem should be implemented as an abstract wrapper, which is aware of this configuration:
```

Comment:
Are validation/pre-validation functions mentioned anywhere else? If not they could go here.

Reply:
Yup, they would go here. However I've deferred that to a later point.